main/src/page-content/content-script.js

```diff
@@ -512,11 +513,18 @@ function getReplaceText(text) {
   // replace regex groups
   const currentOccurrenceText = getCurrentOccurrenceText();
   const matches = Search.lastSearchRegexp.exec(currentOccurrenceText);
+  const replacePlaceholder = (idx) => `◴◶${idx}◵◷`; // This unicode combo is unlikely to occur
   if (matches && matches.length > 0) {
-    // Replace starting from the largest number (replace $11 before $1)
-    matches.slice().reverse().forEach((groupText, index) => {
+    const reversedMatches = matches.slice().reverse();
+    // First stage replace: Replace $0, $1, ... by placeholders
+    reversedMatches.forEach((groupText, index) => {
```

Comment:
Perhaps this could be a single pass. See final review comment.

Reply:
Sorry, but I don't really have time to optimize & debug this right now, as it's a non-essential change IMO. Feel free to create an issue about doing this optimization however - maybe someone will take some time to optimize it in the future.